app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,13 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+      @user = User.from_omniauth(request.env["omniauth.auth"])
+
+      if @user.persisted?
```

Comment:
Adding a note here: When logging in as a user not in the db this chokes. I suspect the answer is altering `User#from_omniauth` a bit so that it looks more like a `find_by email` call.